src/account-abstraction.md

```diff
@@ -0,0 +1,168 @@
+
+
+<b><h3> Enhancing Transaction Flexibility and Security with Account Abstraction </h3></b>
+
+Smart Contracts Paying for Transactions:    
+In the traditional model, users need EOAs to pay gas fees for executing smart contracts. With AA, smart contracts can be designed to handle these fees. This means that the logic within the contract can manage the transaction fees, allowing for more flexible and user-friendly applications. 
+    
+<br>Validation Abstraction:
+	Single Validation Method on Ethereum: On Ethereum Layer 1, transactions are validated using a single method, typically involving the signature of an EOA. This method verifies that the transaction was authorized by the account holder.
+
+Multiple Validation Methods in AA: 
+    AA allows for different types of validations. For instance, instead of just EOA signatures, a contract could accept other cryptographic signatures (like those from multi-signature wallets, hardware wallets, or even different cryptographic algorithms). This flexibility enables diverse and potentially more secure ways to validate transactions. 
+    
+<h3>SNIP-6</h3>
+
+``` #![allow(unused)]
+fn main() {
+/// @title Represents a call to a target contract
+/// @param to The target contract address
+/// @param selector The target function selector
+/// @param calldata The serialized function parameters
+struct Call {
+    to: ContractAddress,
+    selector: felt252,
+    calldata: Array<felt252>
+}
+
+/// @title SRC-6 Standard Account
+trait ISRC6 {
+    /// @notice Execute a transaction through the account
+    /// @param calls The list of calls to execute
+    /// @return The list of each call's serialized return value
+    fn __execute__(calls: Array<Call>) -> Array<Span<felt252>>;
+
+    /// @notice Assert whether the transaction is valid to be executed
+    /// @param calls The list of calls to execute
+    /// @return The string 'VALID' represented as felt when is valid
+    fn __validate__(calls: Array<Call>) -> felt252;
+
+    /// @notice Assert whether a given signature for a given hash is valid
+    /// @param hash The hash of the data
+    /// @param signature The signature to validate
+    /// @return The string 'VALID' represented as felt when the signature is valid
+    fn is_valid_signature(hash: felt252, signature: Array<felt252>) -> felt252;
+}
+
+/// @title SRC-5 Standard Interface Detection
+trait ISRC5 {
+    /// @notice Query if a contract implements an interface
+    /// @param interface_id The interface identifier, as specified in SRC-5
+    /// @return `true` if the contract implements `interface_id`, `false` otherwise
+    fn supports_interface(interface_id: felt252) -> bool;}}
+ ```
+<p> The SNIP-6 interface standardizes account contract should be able to implement ```__execute__```, ```__validate___```,and ```is_valid_signature```</p>
```

Comment:
Change to: Smart wallet contracts **must** implement the `SRC6` and `SRC5` traits.